src/m365/spo/commands/site/site-apppermission-add.ts

```diff
@@ -0,0 +1,123 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  permission: string;
+  appId: string;
+  appDisplayName: string;
+}
+
+class SpoSiteApppermissionAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.SITE_APPPERMISSION_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds an application permissions to the site';
+  }
+
+  private getSpoSiteId(args: CommandArgs): Promise<string> {
+    const url = new URL(args.options.siteUrl);
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/sites/${url.hostname}:${url.pathname}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ id: string }>(requestOptions)
+      .then((site: { id: string }) => site.id);
+  }
+
+  private mapRequestBody(options: Options): any {
+    const applicationInfo: any = {};
+    if (options.appId) {
+      applicationInfo.id = options.appId;
+    }
+
+    if (options.appDisplayName) {
+      applicationInfo.displayName = options.appDisplayName;
+    }
+
+    return `{
```

Comment:
Rather than manually interpolating string, let's define it as an object and then serialize to JSON